core/src/apps/cardano/helpers/paths.py

```diff
@@ -7,21 +7,21 @@
 
 BYRON_ROOT = [44 | HARDENED, SLIP44_ID | HARDENED]
 SHELLEY_ROOT = [1852 | HARDENED, SLIP44_ID | HARDENED]
+MULTISIG_ROOT = [1854 | HARDENED, SLIP44_ID | HARDENED]
+MINTING_ROOT = [1855 | HARDENED, SLIP44_ID | HARDENED]
 
 # fmt: off
-SCHEMA_PUBKEY = PathSchema.parse("m/[44,1852]'/coin_type'/account'/*", SLIP44_ID)
-SCHEMA_ADDRESS = PathSchema.parse("m/[44,1852]'/coin_type'/account'/[0,1,2]/address_index", SLIP44_ID)
+SCHEMA_PUBKEY = PathSchema.parse("m/[44,1852,1854]'/coin_type'/account'/*", SLIP44_ID)
+# minting has a specific schema for key derivation - see CIP-1855
+SCHEMA_MINT = PathSchema.parse("m/1855'/coin_type'/[0-%s]'" % (HARDENED - 1), SLIP44_ID)
+SCHEMA_PAYMENT = PathSchema.parse("m/[44,1852,1854]'/coin_type'/account'/[0,1]/address_index", SLIP44_ID)
```

Comment:
I don't like having the same schema for multisig and ordinary keys. The chain code for multisig can only be 0 or 2, not 1 (unlike ordinary keys).

Reply:
True. Turns out we don't need a multi-sig schema at all, since it's determined only in scripts or witnesses and in both cases the path is always shown. So we only need to check that purpose indeed is 1854.

Reply:
https://github.com/vacuumlabs/trezor-firmware/pull/66/commits/5464dda6bd13dee7dc3c63c36c6024753e8f3c46#diff-1117e5cce47581e08224a93f47952999b3a88dc3177e7d54046ae6f95b583d98R17